src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/SwitchControllerV2.java

```diff
@@ -260,4 +261,23 @@ public CompletableFuture<SwitchValidationResultV2> validateSwitch(
             @RequestParam(name = "exclude", required = false) String excludeString) {
         return switchService.validateSwitch(switchId, includeString, excludeString);
     }
+
+    /**
+     * Retrieves a map of all flows for each port for the given switch.
+     * When a port ID is provided, returns flows only for the given port.
+     * Ports that don't have any associated flow are skipped, i.e. if no flows are going through this switch,
+     * returns an empty output.
+     * @param switchId a specific switch
+     * @param portIds optional. Filters the output to display this port only
+     * @return mapping of port->[flows]
+     */
+    @ApiOperation(value = "Get all flows for each port for the given switch",
+            response = SwitchFlowsPerPortResponse.class)
+    @GetMapping("/{switch_id}/flows-by-port")
```

Comment:
Why do you use an endpoint different to the one from #5015? It's a standard practice in this project to have V1/V2 endpoints with the same name but working slightly different. Please, get back to the initial naming.

Reply:
We discussed it on demo that the pattern of plurals implies that the return type is a collection. Here we have a collection of collections, which is not consistent with other API names, such as connections, properties, and so on. This API in fact does something different that wasn't available in v1. I posted it [in slack](https://miracloud.slack.com/archives/G79HZSL0J/p1672911054415589) for visibility, there were no suggestions, so I changed it @pzakatov suggested